spec/requests/tasks_spec.rb

```diff
@@ -0,0 +1,127 @@
+require 'rails_helper'
+
+RSpec.describe 'Tasks API' do
+
+  let!(:story) { create(:story) }
+  let!(:tasks) { create_list(:task, 20, story_id: story.id) }
+  let(:story_id) { story.id }
+  let(:id) { tasks.first.id }
+
+  # Test suite for GET /stories/:story_id/tasks
+  describe 'GET /stories/:story_id/tasks' do
+    before { get "/stories/#{story_id}/tasks" }
+
+    context 'when story exists' do
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'returns all story tasks' do
+        expect(json.size).to eq(20)
+      end
+    end
+
+    context 'when story does not exist' do
+      let(:story_id) { 0 }
+
+      it 'returns status code 404' do
+        expect(response).to have_http_status(404)
+      end
+
+      it 'returns a not found message' do
+        expect(response.body).to match(/Couldn't find Story/)
+      end
+    end
+  end
+
+  # Test suite for GET /tasks/:id
+  describe 'GET /tasks/:id' do
+    before { get "/tasks/#{id}" }
+
+    context 'when story task exists' do
+      it 'returns status code 200' do
+        expect(response).to have_http_status(200)
+      end
+
+      it 'returns the task' do
+        expect(json['id']).to eq(id)
+      end
+    end
+
+    context 'when story task does not exist' do
+      let(:id) { 0 }
+
+      it 'returns status code 404' do
+        expect(response).to have_http_status(404)
+      end
+
+      it 'returns a not found message' do
+        expect(response.body).to match(/Couldn't find Task/)
+      end
+    end
+  end
+
+  # Test suite for POST /stories/:story_id/tasks
+  describe 'POST /stories/:story_id/tasks' do
```

Comment:
this should be consistent - either we're using the shortest possible route (imo the easier way) or we're using the nested approach - if the former, this should be POST /tasks with story_id as attribute, if the latter all task routes should have /stories/:story_id with them.